### PR TITLE
xds/resolver: cleanup tests to use real xDS client

### PR DESCRIPTION
**Pull request by easwars**
RELEASE NOTES: n/a

### CODE REVIEW

File: xds/internal/resolver/xds_resolver_test.go [hunk 1]
```diff
@@ -197,20 +225,17 @@ func (s) TestResolverBuilder(t *testing.T) {
 			}
 
 			r, err := builder.Build(test.target, newTestClientConn(), test.buildOpts)
-			if (err != nil) != test.wantErr {
+			if gotErr, wantErr := err != nil, test.wantErr != ""; gotErr != wantErr {
```
Comment (arvindbr8) [inline xds/internal/resolver/xds_resolver_test.go:228]: really really nit: At first this seemed a little confusing to me. But it made sense. 
We are trying to check if gotErr and wantErr are either both Nil or not nil. 

can we rename gotErr and wantErr to something like `gotNilErr` and `wantNilErr` ? 

again.. really really nit

File: xds/internal/resolver/xds_resolver_test.go [hunk 2]
```diff
@@ -288,89 +313,128 @@ func waitForWatchRouteConfig(ctx context.Context, t *testing.T, xdsC *fakeclient
...
 		{
-			name: "default %s old style",
-			bc: &bootstrap.Config{
-				ClientDefaultListenerResourceNameTemplate: "%s",
-			},
-			target: resolver.Target{
-				URL: url.URL{Path: "/" + targetStr},
-			},
-			want: targetStr,
+			name:                         "default %s old style",
```
Comment (arvindbr8) [inline xds/internal/resolver/xds_resolver_test.go:332]: nice this is cleaner

**Review by arvindbr8 — APPROVED** [review]
apart from just a really small nit, LGTM!

**Comment by arvindbr8** [comment]
One of these tests `TestResolverResourceName` failed in the GH actions. @easwars

**Comment by easwars** [comment]
> One of these tests `TestResolverResourceName` failed in the GH actions. @easwars

I pushed a small change, allocating the channel with a buffer. This should take care of the flake.

File: xds/internal/resolver/xds_resolver_test.go (hunk 1 above)
Comment (easwars) [reply xds/internal/resolver/xds_resolver_test.go:228]: > We are trying to check if gotErr and wantErr are either both Nil or not nil.

Yes, basically we are trying to verify if we got an error when we expect one, and we dont get an error when we don't expect one. This is a fairly common practice in Go code wherein `gotErr` means `did I get an error?` and `wantErr` means `do i expect to see an error?` here.  It is also common practice for `gotErr` and `wantErr` to also contain the actual error seen by the test and the actual error expected by the test. These are used interchangeably.

I think the more you see this pattern, the more you will feel comfortable with it. 

If I have to rename it as `gotNilErr`, then it actually should be `gotNilErr := (err == nil)`, and that is not what we are doing here. So, I will leave this as is.

**Pull request merged**